ui/src/app/views/dataset-page.spec.tsx

```diff
@@ -94,19 +126,26 @@ describe('DataSet', () => {
     wrapper.find('[data-test-id="value-list-items"]').find('input').first()
       .simulate('change');
 
-    // Buttons should now be enabled
-    const buttons = wrapper.find(Button);
-    expect(buttons.find('[data-test-id="save-button"]').first().prop('disabled'))
-      .toBeFalsy();
+    await waitOneTickAndUpdate(wrapper);
+
+    // Select another value preview data api should not be called now
+    wrapper.find('[data-test-id="value-list-items"]').at(1)
+      .find('input').first().simulate('click');
+
+    // Click preview button to load preview
+    wrapper.find({'data-test-id': 'preview-button'}).first().simulate('click');
+    await waitOneTickAndUpdate(wrapper);
+    expect(spy).toHaveBeenCalledTimes(1);
+
   });
 
-  it('should display preview data only once cohort, concept and value are selected', async() => {
+  it('should refresh preview data once refresh preview link is clicked', async() => {
```

Comment:
nit: can we say refresh preview data table?

Reply:
updated